rules/scripts/sival_signature_test.bash

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -o pipefail
+
+FILES_TO_CHECK=(@FILES@)
+
+IS_NEGATIVE_TEST=@IS_NEGATIVE_TEST@
+OPENTITANTOOL=@OPENTITANTOOL@
+VERIFY_ARGS=(@VERIFY_ARGS@)
+ECDSA_KEY=@ECDSA_KEY@
+SPX_KEY=@SPX_KEY@
+KEY_ARGS=()
+RESULT_STATUS=0
+
+echo "___Starting signature verification & manifest key verification___"
+
+for file in "${FILES_TO_CHECK[@]}"
+do
+
+  echo "Checking signature of ${file}..."
+
+  # TEST 1: ECDSA key and/or SPX key integrity and signature verification
+
+  # If ECDSA key is provided, include it in the manifest update
+  if [[ ! -z "$ECDSA_KEY" ]]; then
+    KEY_ARGS+=("--ecdsa-key=${ECDSA_KEY}")
+  fi
+  
+  # If SPX key is provided, include it in the manifest update
+  if [[ ! -z "$SPX_KEY" ]]; then
+    KEY_ARGS+=("--spx-key=${SPX_KEY}")
+  fi
+
+  # If there are keys to verify, perform manifest update
+  if [[ ${#KEY_ARGS[@]} -ne 0 ]]; then
+    echo -n "Checking key material in ${file}..."
+        ${OPENTITANTOOL}  --rcfile= image manifest update \
+          "${KEY_ARGS[@]}" \
+          --output=/tmp/manifest_test_key_verify.$$ \
+          ${file}
+
+    # Perform manifest comparison
+    echo "Comparing dummy manifest with original file..."
+    cmp ${file} /tmp/manifest_test_key_verify.$$
+
+  fi
+
+  # Check if the comparison was successful
+  if [[ $? -ne 0 ]]; then
+    echo "Manifest compare failed."
+    RESULT_STATUS=1
+  else
+    echo "Manifest compare succeeded."
+  fi
```

Comment:
This needs to be inside the `if/then/fi` above - we should only do this if we performed the manifest update and comparison.